custom/icds/data_management/models.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime
+
+from django.db import models
+from django.utils.translation import ugettext_noop
+
+from custom.icds.data_management.const import DATA_MANAGEMENT_TASKS
+
+
+class DataManagementRequest(models.Model):
+    STATUS_PENDING = 0
+    STATUS_IN_PROGRESS = 1
+    STATUS_SUCCESS = 2
+    STATUS_FAILED = 3
+    STATUS_CHOICES = (
+        (STATUS_PENDING, ugettext_noop('pending')),
+        (STATUS_IN_PROGRESS, ugettext_noop('in progress')),
+        (STATUS_SUCCESS, ugettext_noop('successful')),
+        (STATUS_FAILED, ugettext_noop('failed'))
+    )
+    STATUSES = [STATUS_PENDING, STATUS_IN_PROGRESS, STATUS_SUCCESS, STATUS_FAILED]
+    slug = models.CharField(max_length=255, blank=False, null=False)
+    domain = models.CharField(max_length=255, blank=False, null=False)
+    db_alias = models.CharField(max_length=255, blank=True, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    initiated_by = models.CharField(max_length=255, blank=False, null=False)
+
+    # timestamps of request runtime
+    started_on = models.DateTimeField()
+    ended_on = models.DateTimeField(blank=True, null=True)
+
+    # to consider cases modified within range
+    from_date = models.DateField(blank=True, null=True)
+    till_date = models.DateField(blank=True, null=True)
+
+    error = models.TextField(blank=True, null=True)
+    status = models.SmallIntegerField(choices=STATUS_CHOICES, default=STATUS_PENDING)
+
+    def execute(self):
+        """
+        run the updates through DataManagement
+        """
+        self._set_in_progress()
+        if self.slug not in DATA_MANAGEMENT_TASKS:
+            self._note_error("Unexpected slug %s" % self.slug)
```

Comment:
surely this should be handled earlier

Reply:
`execute` is the only public method/entry point for this class, the form already ensures that no other value is entered in the UI but kept the check on the model as well.
Where before this you suggest it should be checked?

Reply:
fair enough